SPEC.md

```diff
@@ -4854,23 +4877,23 @@ version 1.1
 
 task hisat2 {
   input {
-    File index
+    File index_tar_gz
     String sra_acc
     Int? max_reads
     Int threads = 8
     Float memory_gb = 16
     Float disk_size_gb = 100
   }
 
-  String index_id = basename(index, ".tar.gz")
+  String index_id = basename(index_tar_gz, ".tar.gz")
 
   command <<<
     mkdir index
-    tar -C index -xzf ~{index}
+    tar -C index -xzf ~{index_tar_gz}
     hisat2 \
       -p ~{threads} \
       ~{if defined(max_reads) then "-u ~{select_first([max_reads])}" else ""} \
-      -x index/~{index_id} \
+      -x index/grch38/genome \
```

Comment:
This looks like we're hardcoding for the name of the genome that actually happens to be in this file; we also no longer use `index_id` at all.

Maybe we meant to generate this string from the filename?

```suggestion
      -x index/"$(echo ~{index_id} | tr '_' '/')" \
```

Reply:
I'm guessing the tgz file contains the reference nested two levels deep. Rather than hard-code the path, I used the `--strip-components 2` argument to `tar`.